File: java/object-oriented programming/interface/mavenproject4/Funcionario.java
package com.mycompany.mavenproject4;

public abstract class Funcionario {
    protected String nome;
    protected String dataNascimemto;
    protected Sexo sexo;
    protected Setor setor;
    protected double salarioBase;

    public Funcionario(String nome, String dataNascimemto, Sexo sexo, Setor setor, double salarioBase) {
        this.nome = nome;
        this.dataNascimemto = dataNascimemto;
        this.sexo = sexo;
        this.setor = setor;
        this.salarioBase = salarioBase;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getDataNascimemto() {
        return dataNascimemto;
    }

    public void setDataNascimemto(String dataNascimemto) {
        this.dataNascimemto = dataNascimemto;
    }

    public Sexo getSexo() {
        return sexo;
    }

    public void setSexo(Sexo sexo) {
        this.sexo = sexo;
    }

    public Setor getSetor() {
        return setor;
    }

    public void setSetor(Setor setor) {
        this.setor = setor;
    }

    public double getSalarioBase() {
        return salarioBase;
    }

    public void setSalarioBase(double salarioBase) {
        this.salarioBase = salarioBase;
    }

    public abstract double getSalarioFinal();
    
    @Override
    public String toString() {
        return "\nNome: " + nome + 
               "\nData de Nascimemto: " + dataNascimemto + 
               "\nSexo: " + sexo.getTexto() + 
               "\nSetor: " + setor.getNome() + 
               "\nSalário Base: " + salarioBase +
               "\nSalário Final: " + getSalarioFinal();
    }
}

File: python/projects/project 02/models/funcionario.py
from models.enums.sexo import Sexo
from models.enums.setor import Setor

class Funcionario:
    def __init__(self, id: int, nome: str, idade: int, salario: float, setor: Setor, sexo: Sexo) -> None:
        self.id = id
        self.nome = nome
        self.idade = idade
        self.salario = salario
        self.setor = setor
        self.sexo = sexo

    def __str__(self) -> str:
        return (
                f"Id: {self.id}"
                f"\nNome: {self.nome}"
                f"\nIdade: {self.idade}"
                f"\nSalário: {self.salario}"
                f"\nSetor: {self.setor.value}"
                f"\nSexo: {self.sexo.value}"
                )